### ISSUE TITLE
AnyDateTimeFormat should fallback calendars by locale

### ISSUE THREAD

**Issue by Manishearth** [unicode-org/icu4x#2038]
Blocked on https://github.com/unicode-org/icu4x/issues/1109

https://github.com/unicode-org/icu4x/pull/1987 adds formatting for AnyCalendar, however it unconditionally falls back to the Gregorian calendar. We should instead be getting this fallbacking from data, _ideally_ using vertical fallback (but we can just have a `calendars/default@1` key if we really have to).

Filing a separate issue so it can be a polish blocker.

**Comment by sffc** [comment]
The regions using non-Gregorian default are:

- AF, IR (persian)
- SA (islamic-umalqura)
- TH (buddhist)

Source: https://github.com/unicode-org/cldr/blob/b2b36963947b0e3e172fa40519d13afd9877a227/common/supplemental/supplementalData.xml#L4715

Of those, the only one we support right now is Buddhist. So to fix the 1.0 issue, we can hard-code a check for TH. Given that Thai is one of our testdata languages, this would also be a nice test to add to demonstrate the i18n of DateTimeFormatter.

**Comment by sffc** [comment]
Created #2236 for the longer-term solution to this problem.

**Comment by Manishearth** [comment]
Related: https://github.com/unicode-org/icu4x/issues/2357
